File: csharp/space-age/SpaceAge.cs
using System;

public class SpaceAge
{
    private int _earthSeconds = 0;
    public SpaceAge(int seconds) => _earthSeconds = seconds;

    public double OnEarth() => _earthSeconds / 31557600.0;

    public double OnMercury() => this.OnEarth() / 0.2408467;

    public double OnVenus() => this.OnEarth() / 0.61519726;

    public double OnMars() => this.OnEarth() / 1.8808158;

    public double OnJupiter() => this.OnEarth() / 11.862615;

    public double OnSaturn() => this.OnEarth() / 29.447498;

    public double OnUranus() => this.OnEarth() / 84.016846;

    public double OnNeptune() => this.OnEarth() / 164.79132;
}

File: scala/space-age/src/main/scala/SpaceAge.scala
object SpaceAge {

  def onEarth(secondAge: Double): Double = secondAge/31557600.0
  def onMercury(secondAge: Double): Double = onEarth(secondAge) / 0.2408467
  def onVenus(secondAge: Double): Double = onEarth(secondAge) / 0.61519726
  def onMars(secondAge: Double): Double = onEarth(secondAge) / 1.8808158
  def onJupiter(secondAge: Double): Double = onEarth(secondAge) / 11.862615
  def onSaturn(secondAge: Double): Double = onEarth(secondAge) / 29.447498
  def onUranus(secondAge: Double): Double = onEarth(secondAge) / 84.016846
  def onNeptune(secondAge: Double): Double = onEarth(secondAge) / 164.79132
}
